### PR TITLE
Restructure build system to use Taskfile

### PR DESCRIPTION
**Pull request by MDr164**
This PR will be a bit bigger. I completely restructured the way u-bmc is built and made the process more modular. With this approach it'll be easier to add future platforms and new SoCs to the source. The Go code remained untouched for now. It's also easier to track down errors in the build system and it's less fragile compared to GNU make in my opinion. This PR also requires #215 and solves #203 and also lays the ground to solve #204

### CODE REVIEW

File: taskfiles/Proto.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+tasks:
+  # Update gRPC protocol definitions
+  protogen:
+    run: once
+    dir: proto
+    cmds:
+      - protoc --go_out=plugins=grpc:. bmc.proto
+      - protoc --go_out=plugins=grpc:. config.proto
+
```
Comment (ChriMarMe) [inline taskfiles/Proto.yml:11]: Remove the newline at end of file.

**Review by ChriMarMe — DISMISSED** [review]

**Comment by ChriMarMe** [comment]
Good work! :+1:

File: taskfiles/Proto.yml (hunk 1 above)
Comment (MDr164) [reply taskfiles/Proto.yml:11]: I think I was generally a bit inconsistent with newlines at the end of the Taskfiles, I'll clean that up later today then.

Comment (ChriMarMe) [reply taskfiles/Proto.yml]: My perception was wrong. Somehow the complaint was "No newline at end of file" and I interpreted that as "remove the line" Seems like it was complaining about the missing of it.

**Review by ChriMarMe — APPROVED** [review]

**Pull request merged**